lib/Reader/readerir.cpp

```diff
@@ -2069,20 +2094,29 @@ IRNode *GenIR::convertFromStackType(IRNode *Node, CorInfoType CorType,
   }
 
   case Type::TypeID::PointerTyID: {
+    bool PointerRepresentsStruct = doesValueRepresentStruct(Node);
+    if (PointerRepresentsStruct) {
+      if (Ty->getPointerElementType() != ResultTy) {
```

Comment:
I would add an assert here that `ResultTy` is a `StructType` and that the layout of the two structs is the same (similar to the merge case). We don't want to reinterpret a struct's layout here.